File: src/main/webapp/service/businessPartnerService.js
/**
 * Created by JELENA on 3.6.2017.
 */

app.service('businessPartnerService', function($http) {
    return {
        read: function (onSuccess, onError) {
            $http.get('api/businessPartners').then(onSuccess, onError);
        },
        create: function(businessPartner, onSuccess, onError) {
            $http.post('/api/businessPartners', businessPartner).then(onSuccess, onError);
        },
        update: function (businessPartner, onSuccess, onError) {
            $http.patch('/api/businessPartners/' + businessPartner.id, businessPartner).then(onSuccess, onError);
        },
        delete: function (id, onSuccess, onError) {
            $http.delete('/api/businessPartners/' + id).then(onSuccess, onError);
        }
    }
});

File: src/main/java/com/ftn/service/BusinessPartnerService.java
package com.ftn.service;

import com.ftn.model.BusinessPartner;
import com.ftn.model.Warehouse;

import java.util.List;

/**
 * Created by Olivera on 21.5.2017..
 */
public interface BusinessPartnerService {

    List<BusinessPartner> read();

    List<BusinessPartner> readByCompany(Long id);

    BusinessPartner create(BusinessPartner businessPartner);

    BusinessPartner update(Long id, BusinessPartner businessPartner);

    void delete(Long id);
}
